src/app/features/ryder/ryder-utils.ts

```diff
@@ -208,6 +208,12 @@ class RyderApp {
     callback: (res: Response<boolean>) => void
   ): Promise<void> {
     this.ryder_serial = new RyderSerial(payload.port, payload.options);
+    let wrapped_callback = (res) => {
+      // The `RyderSerial` must be closed before invoking the callback, or the latter might create a
+      // new `RyderSerial` and get a device busy error
```

Comment:
When did this happen? Can we not check if it is already open and just keep it if so?

Reply:
That should be possible and would be a better solution I think. However, I am unable to reproduce the original issue here. I remember inspecting the log and finding that instances were being created before the previous one was closed, leading to device busy errors, but it doesn't seem to be happening anymore.

Reply:
If the callback makes another call to Ryder, the previous RyderSerial might be still open. It is a good idea wrap the callback and help developers to do be able to call to Ryder from within the callback.

Reply:
Ah understood. Yes that would be a problem with the sequencer, it should deal with all those issues.

Reply:
It would be easiest to make sure any user of RyderSerial only ever opens a single instance. Then, multiple commands can be sent in a sequence without issue:
```typescript
let ryder = new RyderSerial(...);
let a = await ryder.send(...);
let b = await ryder.send(...);
```
This will require changes to `ryder-utils`, but I tried keep changes to it to a minimum in this PR.